tst/unit/test_unit_params.cpp

```diff
@@ -39,41 +39,33 @@ TEST_CASE("Add and Get is called", "[Add,Get]") {
       REQUIRE_THROWS_AS(params.Get<int>(key), std::invalid_argument);
     }
   }
-
-  GIVEN("An empty params structure") {
-    Params params;
-    WHEN(" attempting to get a key that does not exist ") {
-      std::string non_existent_key = "key";
-      REQUIRE_THROWS_AS(params.Get<double>(non_existent_key), std::invalid_argument);
```

Comment:
Are you no longer checking what happens when accessing a key that no longer exists?

Reply:
I can't test for this, because `PARTHENON_REQUIRE` causes the code to die with a useful error message but without throwing an error. I could replace the `PARTHENON_REQUIRE` call with an error throw if that's preferred. Then this test could be added again.

Reply:
If it's a performance issue I would leave it as it is. I tend to use exceptions, if it a class the end user is likely to have some kind of access too (I would expect the params class would fit into this category), and assertion type checks for internal checks. However, because of the performance issues with exceptions I keep getting corrected on my suggestions related to this issue :). I was more just curious.

Reply:
I don't think this is a performance-critical piece of code. So exceptions are probably fine here. I can change it back. The code will still be cleaner than it was.

Reply:
I would see what @nmsriram says, before doing any more work, as it is, it is an improvement.

Reply:
I agree with @Yurlungur that this is not a performance critical piece of code, since it should only be done once per physics.  Testing things that kill the code are always problematic and I'm not sure I know of any testing frameworks that handle it better.  

Is it possible to capture this by using std::terminate functionality so that the Kokkos abort would eventually end up in a function we designate?

Reply:
Could we address this with https://cmake.org/cmake/help/v3.7/prop_test/WILL_FAIL.html ?
Overall, I'm generally in favor of adding test rather than removing ;)

Reply:
@pgrete @JoshuaSBrown @nmsriram I don't know what the performance implications of `std::terminate` are... and at a cursory pass, I don't see how to get `WILL_FAIL` to behave properly with `Catch2`. How about instead I incorporate error throwing into error checking? I.e., I add a new `PARTHENON_REQUIRE_THROWS` and `PARTHENON_FAIL_THROWS` which throws errors?

Reply:
@Yurlungur I'm in favor of your suggestion.

Reply:
@JoshuaSBrown @nmsriram @pgrete can you take another look? I've added the ability to throw in `error_checking.hpp`.